testfixtures/logcapture.py

```diff
@@ -73,11 +84,54 @@ def atexit(cls):
                 '%s' % ('\n'.join((str(i.names) for i in cls.instances)))
                 )
 
+    def __len__(self):
+        return len(self.records)
+
+    def __getitem__(self, index):
+        return self._actual_row(self.records[index])
+
+    def __contains__(self, what):
+        for i, item in enumerate(self):
+            if what == item:
+                self.records[i].checked = True
+                return True
+
     def clear(self):
         """Clear any entries that have been captured."""
         self.records = []
 
+    def mark_all_checked(self):
+        """
+        Mark all captured events as checked,
+        use this when doing the assertion externally
+        """
+        for record in self.records:
+            record.checked = True
+
+    def ensure_checked(self, level=None):
+        """
+        Ensure every entry logged above the specified `level` has been checked.
+        Raises an :class:`AssertionError` if this is not the case.
+
+        :param level: the logging level, defaults to :attr:`ensure_checks_above`.
+        :type level: Union[None, int]
+        """
+        if level is None:
+            level = self.ensure_checks_above
+        if level == logging.NOTSET:
+            return
+        un_checked = []
+        for record in self.records:
+            if level <= record.levelno and not record.checked:
```

Comment:
Never seen "greater than" expressed this way round before, let's change it:
```suggestion
            if record.levelno >= level and not record.checked:
```